src/DiagramGenerator/Diagram.php

```diff
@@ -150,7 +150,8 @@ public function draw()
             $this->image = $this->image->appendImages(true);
         }
 
-        $this->image->setImageFormat('png');
+        $this->image->setImageFormat($this->config->getImageFormat());
+        $this->image->setImageCompressionQuality($this->config->getImageCompressionQuality());
```

Comment:
Does this setting affect PNGs? I had worked with a previous version of Image Magick where if you set the JPEG compression quality on a non-JPEG, you would get an error.